cloudinit/net/netplan.py

```diff
@@ -235,6 +237,61 @@ def _clean_default(target=None):
         os.unlink(f)
 
 
+def netplan_api_write_yaml_file(net_config_content: str) -> bool:
+    """Use netplan.State._write_yaml_file to write netplan config
+
+    Where netplan python API exists, prefer to use of the private
+    _write_yaml_file to ensure proper permissions and file locations
+    are chosen by the netplan python bindings in the environment.
+
+    By calling the netplan API, allow netplan versions to change behavior
+    related to file permissions and treatment of sensitive configuration
+    under the API call to _write_yaml_file.
+
+    In future netplan releases, security-sensitive config may be written to
+    separate file or directory paths than world-readable configuration parts.
+    """
+    try:
+        from netplan.parser import Parser
+        from netplan.state import State
+    except ImportError:
+        LOG.debug(
+            "No netplan python module. Fallback to write %s",
+            CLOUDINIT_NETPLAN_FILE,
+        )
+        return False
+    cfg_dir = mkdtemp()
+    try:
+        with open(os.path.join(cfg_dir, "ci-net-cfg.yaml"), "wb+") as f:
+            f.write(util.encode_text(net_config_content))
```

Comment:
note (non-blocking): You could consider doing this all in-memory, using `io.StringIO` and/or `os.memfd_create`

Reply:
Or `tempfile.SpooledTemporaryFile`